src/main/java/net/codejava/MvcController.java

```diff
@@ -522,4 +540,178 @@ public String submitDispute(@ModelAttribute("user") User user) {
+   * 
+   * If the password attempt is correct, the sum of money is deposited
+   * and savings balance is updated
+   * 
+   * If the password attempt is incorrect, the user is redirected to the "welcome" page.
+   * 
+   * @param user
+   * @return "account_info" page if login successful. Otherwise, redirect to "welcome" page.
+   */
+  @PostMapping("/withdraw_savings")
+  public String submitWithdrawSavings(@ModelAttribute("user") User user)  {
+    String userID = user.getUsername();
+    String userPasswordAttempt = user.getPassword();
+    dt = new java.util.Date();
+    sdf = new java.text.SimpleDateFormat("yyyy-MM-dd HH:mm:ss");
+    SQL_DATETIME_FORMAT = sdf.format(dt);
+    double depositAmount = 0;
+
+    String getUserPasswordSql = String.format("SELECT Password FROM passwords WHERE CustomerID='%s';", userID);
+    String userPassword = jdbcTemplate.queryForObject(getUserPasswordSql, String.class);
+
+    // unsuccessful login
+    if (userPasswordAttempt.equals(userPassword) == false) {
+      return "welcome";
+    }
+
+
+    double userWithdrawAmt = user.getAmountToWithdraw();
+    int userWithdrawAmtInPennies = (int) (userWithdrawAmt * 100);
+    
+    if (userWithdrawAmt < 0){
+      return "welcome";
+    }
+
+    // how many times to apply interest
+    int interestCount = 0;    
+
+    // check if date of the savings input passed seven days, and if amounts are equal
+    System.out.println( "WITHDRAW AMOUNT: " + userWithdrawAmtInPennies);
+    String getSavingsAmount = String.format("SELECT Amount FROM savings WHERE CustomerID='%s' AND Amount = %d;", 
+    userID, userWithdrawAmtInPennies);
+    int savingsAmount = jdbcTemplate.queryForObject(getSavingsAmount, Integer.class);
+
+    String getSavingsDate = String.format("SELECT Timestamp FROM savings WHERE CustomerID='%s' AND Amount = %d;", 
+    userID, userWithdrawAmtInPennies);
+    String savingsDate = jdbcTemplate.queryForObject(getSavingsDate, String.class);
+
+
+    // if the withdraw request matches the savings deposit
+    if(savingsAmount == userWithdrawAmtInPennies){
+      // parsing the numbers to create calendar objects for comparison
+      int savingsYear = Integer.parseInt(savingsDate.substring(0,4));
+      int savingsMonth = Integer.parseInt(savingsDate.substring(5,7));
+      int savingsDay = Integer.parseInt(savingsDate.substring(8,10));
+      int currYear = Integer.parseInt(SQL_DATETIME_FORMAT.substring(0,4));
+      int currMonth = Integer.parseInt((SQL_DATETIME_FORMAT.substring(5,7)));
+      int currDay = Integer.parseInt(SQL_DATETIME_FORMAT.substring(8,10));
+
+      Calendar savingsCal = new  GregorianCalendar(savingsYear, savingsMonth, savingsDay);
+      Calendar currCal = new  GregorianCalendar(currYear, currMonth, currDay);
+      
+      long numDays = ChronoUnit.DAYS.between(savingsCal.getTime().toInstant(), currCal.getTime().toInstant());
+      System.out.println(numDays);
+
+      interestCount+=numDays/7;

```

Comment:
No way for me to test your code because I would have to wait 7 days. I would recommend changing this to 7 seconds so that you and I can easily test the feature as we develop it.

Reply:
I will do that.